### DISCUSSION TITLE
Is it possible to use timestamp queries in any browsers?

### DISCUSSION THREAD

**Discussion by OmarShehata** [gpuweb/gpuweb#3354] [Q&A] [answered]
Currently trying to use timestamp queries in Chrome Canary (Version 106.0.5249.0 (Official Build) canary (64-bit)) produces the following error:

```
DOMException: Disallowed feature requested.
```

Requesting it like this:

```
const device = await adapter.requestDevice({
    requiredFeatures: ["timestamp-query"],
  });
```

I'm aware it's optional in the WebGPU spec, and disabled for security considerations (https://github.com/gpuweb/gpuweb/issues/2218). I found a discussion here (https://github.com/gpuweb/gpuweb/issues/2046#issuecomment-918799989) about supporting timestamp queries behind a command line flag so that developers can use this when developing, but I haven't been able to find any information on if such a flag exists. 

Does anyone know the current status of that?

**Comment by gyagp** [comment, marked as answer]
You may pass "--disable-dawn-features=disallow_unsafe_apis" to Chrome and it works there.

**Reply by OmarShehata** [reply]
Awesome, thank you so much!

**Reply by OmarShehata** [reply]
I put together a short article on how to use timestamp queries in case anyone finds it helpful to see a full example: https://github.com/OmarShehata/webgpu-compute-rasterizer/blob/main/how-to-use-timestamp-queries.md
